### PR TITLE
drivers: pinctrl: pfc_rcar: add support of voltage control to pfc driver

### PR DESCRIPTION
**Pull request by xakep-amatop**
Add support of voltage control to Renesas PFC driver. Voltage register mappings have been added to r8a77951 and r8a77961 SoCs.

Allow 'power-source' property for 'renesas,rcar-pfc' node. This property will be used for configuring IO voltage on appropriate pin. For now it is possible to have only two voltages: 1.8 and 3.3.

Note: it is possible to change voltage only for SD/MMC pins on r8a77951 and r8a77961 SoCs.

### CODE REVIEW

File: soc/arm64/renesas_rcar/Kconfig [hunk 1]
```diff
@@ -12,6 +12,10 @@ config SOC_FAMILY
 	string
 	default "renesas_rcar"
 
+config PIN_VOLTAGE_CONTROL
+	bool
+	default y
+
```
Comment (gmarull) [inline soc/arm64/renesas_rcar/Kconfig:18]: this name is too generic, please namespace this

**Review by gmarull — CHANGES_REQUESTED** [review]

File: soc/arm64/renesas_rcar/Kconfig (hunk 1 above)
Comment (gmarull) [reply soc/arm64/renesas_rcar/Kconfig:18]: also, this looks like a pinctrl driver option, why in soc/?

**Review by aaillet — CHANGES_REQUESTED** [review]
Thanks for the PR
Why would this feature be an option ?
Please wait for #56043 before merging this in order for it not to be updated at each PR merge.

**Comment by xakep-amatop** [comment]
@gmarull 
> this name is too generic, please namespace this

done

> also, this looks like a pinctrl driver option, why in soc/?

you're right, done!

-----
@aaillet
> Please also add the feature and pin list to the arm(32) side of the soc !

added!

> Why would this feature be an option ?

To be able to disable this feature for debugging purposes, and with the config, we don't need to care about the existence of PFC SoC changes for other boards (like ARM32 or Gen4).

> Please wait for https://github.com/zephyrproject-rtos/zephyr/pull/56043 before merging this in order for it not to be updated at each PR merge.

Okay, I'll wait, but I don't see any major merging conflicts between these two PRs.

**Pull request closed without merging**